scripts/test-many-configs.sh

```diff
@@ -0,0 +1,419 @@
+        cd ${currModule}
+    fi
+    mkdir .dtfixingtools
+    if [[ ${currModule} == "" ]]; then
+        cp ${scriptDir}/original-order-files/${projName} .dtfixingtools/original-order
+    else
+        cp ${scriptDir}/original-order-files/${currModule} .dtfixingtools/original-order
+    fi
+    cd ${projectDirectory}
+}
+
+
+
+function cleanUp() {
+    testNum=$1
+    currModule=$2
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ".dtfixingtools" ]]; then
+        cd ${projectDirectory}
+        return 0
+    fi
+    cp -r .dtfixingtools ${testNum}
+    rm -r .dtfixingtools
+    cd ${projectDirectory}
+}
+
+
+
+function checkDetType() {
+    testNum=$1
+    detectType=$2
+    currModule=$3
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ${testNum} ]]; then
+        cd ${projectDirectory}
+        return 0
+    fi
+    cd ${testNum}
+    cd detection-results
+    if [[  -d ${detectType} ]]; then
+        echo "Success: Flaky tests were found using the ${detectType} detector type."
+        cd ${projectDirectory}
+        return 0
+    fi
+    echo "No flaky tests were found using the ${detectType} detector type."
+    cd ${projectDirectory}
+    return 1
+}
+
+
+
+function checkNumberRounds() {
+    testNum=$1
+    expectedRounds=$2
+    detectType=$3
+    expectedBehavior=$4
+    currModule=$5
+    if [[ ${currModule} != "" ]]; then
+        cd ${currModule}
+    fi
+    if [[ ! -d ${testNum} ]]; then
+        cd ${projectDirectory}
+        return 0
+    fi
+    cd ${testNum}
+    cd detection-results
+    if [[  -d ${detectType} ]]; then
+        cd ${detectType}
+        numRounds=$(ls | wc -l)
+        if [ ${numRounds} ${expectedBehavior} ${expectedRounds} ]; then
+            echo "Success: ${testNum} ran ${numRounds} round(s) as expected."
+            cd ${projectDirectory}
+            return 0
+        else
+            echo "Error: ${testNum} ran ${numRounds} round(s) when it should have ran ${expectedRounds}. %%%%%"
+            flag=1
+            cd ${projectDirectory}
+            return 1
+        fi
+    fi
+    cd ${projectDirectory}
+    echo "No flaky tests were found using ${expectedRounds} rounds."
+}
+
+
+
+function checkAbsPath() {
+    path=$1
+    resultsFolder=$2
+    projName=$3
+    currModule=$4
+    cd ${path}
+    if [[ -d ${resultsFolder} ]]; then
+        cd ${resultsFolder}
+        if [[ ${currModule} != "" ]]; then
+            cd ${currModule}
+        else
+            cd ${projName}
+        fi
+        if [ -d "detection-results" ]; then
+            echo "Success: dt.cache.absolute.path configuration works as intended."
+            return 0
+        fi
+    fi
+    echo "Error: dt.cache.absolute.path not working as intended. %%%%%"
+    flag=1
+    return 1
+}
+
+
+
+function flakyTestsFound() {
+    testNum=$1
+    expectedTests=$2
+    currModule=$3
+    if [[ $expectedTests == -* ]]; then
```

Comment:
Put a quick comment about how having a "-" as a prefix for this entity means expecting some error message.